File: BankBankApp.Interfaces/IAccount.vb
Imports BankBankApp.BO

Public Interface IAccount
    Inherits ICrud(Of Account)

    Function GetByUserID(userID As Integer) As List(Of Account)

    Function GetByAccountNumber(accountNumber As String) As Account

    Function GetByCardNumber(cardNumber As String) As Account

    Function UpdateAccountStatus(accountID As Integer, status As String) As Integer

End Interface

File: BankBankApp.DAL/Interfaces/IAccount.cs
﻿using System;
using System.Collections.Generic;
using System.Text;
using BankBankApp.BO;

namespace BankBankApp.DAL.Interfaces
{
    public interface IAccountDAL : ICrud<Account>
    {
        Account GetByAccountNumber(string accountNumber);
    }
}
